cloudant-client/src/test/java/com/cloudant/tests/CloudantClientTests.java

```diff
@@ -114,8 +114,9 @@ public void cookieTest() {
      * Assert that the User-Agent header is of the expected form.
      */
     @Test
-    public void testUserAgentHeaderString() {
-        String userAgentHeader = new LibraryVersion().getUserAgentString();
+    public void testUserAgentHeaderString() throws Exception {
+
+        String userAgentHeader = new UserAgentInterceptor(UserAgentInterceptor.class.getClassLoader(), "META-INF/client.properties").getUserAgent();

```

Comment:
I think this could use a comment to point out that the file isn't actually present on the test classloader so it is validating the "unknown" case.
As discussed a second test case that uses a classloader with some test file would be valuable.